client-app/src/desktop/common/Wrapper.js

```diff
@@ -20,10 +68,42 @@ class Wrapper extends Component {
                     marginBottom: 10,
                     omit: !description
                 }),
-                children
+                children,
+                dockContainer({model: this.dockContainerModel})
             ],
             ...rest
         });
     }
+    
+    renderLinks(props) {
```

Comment:
once you change to call `super(props)` in the constructor, you won't need to pass the props through as method args like this, they will get installed on the object, and available in all methods....